### PR TITLE
add cdc types

### PR DESCRIPTION
**Pull request by weekface**
### What problem does this PR solve? 

Add [ticdc](https://pingcap.com/docs/stable/reference/tools/ticdc/deploy/) types. Part of https://github.com/pingcap/tidb-operator/issues/2056

### Check List 

Tests 

Code changes

 - Has Go code change

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Support TiCdc in TidbCluster CR.
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -335,6 +339,25 @@ type TiFlashSpec struct {
 	LogTailer *LogTailerSpec `json:"logTailer,omitempty"`
 }
 
+// TiCdcpec contains details of TiCdc members
+// +k8s:openapi-gen=true
+type TiCdcSpec struct {
```
Comment (Yisaer) [inline pkg/apis/pingcap/v1alpha1/types.go:344]: Do we need to add a `TidbClusterRef` to indicate this TiCDC's target tidbcluster?

Comment (weekface) [reply pkg/apis/pingcap/v1alpha1/types.go:344]: TiCdcSpec is part of `TidbCluster`, so it doesn't need the `TidbClusterRef ` attribute.

**Comment by DanielZhangQD** [comment]
Can we change `Cdc` to `CDC` in the variable and function names? It's a little weird with `Cdc`.

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 2]
```diff
@@ -339,13 +339,13 @@ type TiFlashSpec struct {
 	LogTailer *LogTailerSpec `json:"logTailer,omitempty"`
 }
 
-// TiCdcpec contains details of TiCdc members
+// TiCDCpec contains details of TiCDC members
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/types.go:342]:
```suggestion
// TiCDCSpec contains details of TiCDC members
```

**Pull request merged**